### ISSUE TITLE
Resize Second Derivatives

### ISSUE THREAD

**Issue by 63617020626164** [tensorflow/tensorflow#38643]
**System information**
- TensorFlow version (you are using): 2.2.0.dev20200416
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**

`tf.image.resize` can calculate derivatives but there exists no gradient for the gradient of it.

**Will this change the current api? How?**

No (the API will not change, but the higher-order derivative will be calculable)

**Who will benefit with this feature?**

I ran into this issue when implemented ProGAN with a regularized JS-GAN, as it requires the second derivative of the discriminator, which involves a resize operation. If a second-order derivative is added, this will be possible.

**Any Other info.**

N/A

**Comment by 63617020626164** [comment]
You can get around this by using `tf.nn.avg_pool2d`, which has a second derivative, though it means you can't use other resize methods.

**Comment by 63617020626164** [comment]
For upsampling by 2x, this code works (for `NCHW`):

```python
_upsample_matrix: tf.Tensor = tf.ones([2, 2, 1, 1])
def upsample(x: tf.Tensor) -> tf.Tensor:
    batch, channels, height, width = x.shape
    x = tf.reshape(x, [-1, 1, height, width])
    x = tf.nn.conv2d_transpose(
        x, _upsample_matrix, (height * 2, width * 2), 2, data_format='NCHW'
    )
    x = tf.reshape(x, [-1, channels, height * 2, width * 2])
    return x
```

Implementing this using `NHWC` needs `tf.transpose`, which is less efficient:

```python
_upsample_matrix: tf.Tensor = tf.ones([2, 2, 1, 1])
def upsample(x: tf.Tensor) -> tf.Tensor:
    x = tf.transpose(x, perm=[0, 3, 1, 2])
    batch, channels, height, width = x.shape
    x = tf.reshape(x, [-1, 1, height, width])
    x = tf.nn.conv2d_transpose(
        x, _upsample_matrix, (height * 2, width * 2), 2, data_format='NCHW'
    )
    x = tf.reshape(x, [-1, channels, height * 2, width * 2])
    x = tf.transpose(x, perm=[0, 2, 3, 1])
    return x
```

However both these workarounds are still limited to nearest neighbour(ish) upsampling.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
